src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -753,6 +758,55 @@ public static ServerHostInfo GetServerHostInfo(IDbConnection connection)
             return hostInfo;
         }
 
+         /// <summary>
+        /// Gets the server host cpu count and memory from sys.dm_os_sys_info view
+        /// </summary>
+        /// <param name="connection">The connection</param>
+        public static ServerSysInfo GetServerSysInfo(IDbConnection connection)
+        {
+            var sysInfo = new ServerSysInfo();
+            // SQL servers prior to 2012 provide memory in bytes and not kilobytes.
+            if (!Version.TryParse(ReadServerVersion(connection), out var hostVersion) || hostVersion.Major < 11)
+            {
+                try
+                {
+                    ExecuteReader(
+                        connection,
+                        SqlConnectionHelperScripts.GetHostSysInfoBeforeVersion11,
+                        reader =>
+                        {
+                            reader.Read();
+                            sysInfo.CpuCount = Int32.Parse(reader[0].ToString(), CultureInfo.InvariantCulture);
+                            sysInfo.PhysicalMemoryInKb = long.Parse(reader[1].ToString(), CultureInfo.InvariantCulture)/1024;
```

Comment:
Try to use SMO because it also knows about the special case for SQL Managed Instance, which has data in sys.server_resource_stats instead of dm_os_host_info